addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,597 @@
+-- Define default values ------------------------------------------------------
+-------------------------------------------------------------------------------
+defaults = T{}
+defaults.window = {}
+
+-------------------------------------------------------------------------------
+-- Load defaults from settings.xml --------------------------------------------
+-------------------------------------------------------------------------------
+settings = config.load(defaults)
+
+-- upgrade to new settings file if old settings still exist -------------------
+if settings.trposx and settings.trposy then
+    -- copy position from old config to new one
+    settings.window.pos.x = settings.trposx
+    settings.window.pos.y = settings.trposy
+    -- delete old settings here -----------------------------------------------
+    config.save(settings, 'all')
+end
 
-staggers['day']['random'] = {"Nightmare Taurus"}
-staggers['day']['none'] = {"Animated Claymore", "Animated Dagger", "Animated Great Axe", "Animated Gun", "Animated Hammer", "Animated Horn", "Animated Kunai", "Animated Knuckles", "Animated Longbow", "Animated Longsword", "Animated Scythe", "Animated Shield", "Animated Spear", "Animated Staff", "Animated Tabar", "Animated Tachi", "Fire Pukis", "Petro Pukis", "Poison Pukis", "Wind Pukis", "Kindred's Vouivre", "Kindred's Wyvern", "Kindred's Avatar", "Vanguard Eye", "Prototype Eye", "Nebiros's Avatar", "Haagenti's Avatar", "Caim's Vouivre", "Andras's Vouivre", "Adamantking Effigy", "Avatar Icon", "Goblin Replica", "Serjeant Tombstone", "Zagan's Wyvern", "Hydra's Hound", "Hydra's Wyvern", "Hydra's Avatar", "Rearguard Eye", "Adamantking Effigy", "Adamantking Image", "Avatar Icon", "Avatar Idol", "Effigy Prototype", "Goblin Replica", "Goblin Statue", "Icon Prototype", "Manifest Icon", "Manifest Icon", "Prototype Eye", "Serjeant Tombstone", "Statue Prototype", "Tombstone Prototype", "Vanguard Eye", "Vanguard's Avatar", "Vanguard's Avatar", "Vanguard's Avatar", "Vanguard's Avatar", "Vanguard's Crow", "Vanguard's Hecteyes", "Vanguard's Scorpion", "Vanguard's Slime", "Vanguard's Wyvern", "Vanguard's Wyvern", "Vanguard's Wyvern", "Vanguard's Wyvern", "Warchief Tombstone"}
+Green = "\\cs(0,255,0)"
+Red = "\\cs(255,0,0)"
+Yellow = "\\cs(255,255,0)"
+
+current_mob = nil
+current_proc = nil
+obj_time = 0
+end_time = 0
+
+window = texts.new(" ",settings.window,settings)
+w = T{}
+
+-------------------------------------------------------------------------------
+-- Initialize the Currency array. We need this to keep track of the drops -----
+-------------------------------------------------------------------------------
+function init_currency()
+    Currency = T{"Ordelle Bronzepiece","Montiont Silverpiece","One Byne Bill",
+                "One Hundred Byne Bill","Tukuku Whiteshell","Lungo-Nango Jadeshell",
+                "Forgotten Thought","Forgotten Hope","Forgotten Touch","Forgotten Journey","Forgotten Step"}
```

Comment:
Since you're using T-tables you can use their advanced iteration features:
```lua
    for c in Currency:it() do
        w[c] = 0
    end
```

Same goes for the `init_granules` function below.